src/components/SimpleList/helpers/formatString.js

```diff
@@ -0,0 +1,70 @@
+import React, { Component } from 'react';
+import { wrapTooltip } from 'components/ToolTip';
+import { formatCurrency, formatDate } from './formatters';
+
+/**
+ * Detects various keywords instructing the formatter to parse as Date, Currency, SVG or Bolded
+ * keywords:
+ * \:d and optionally prepend |<Date Format> - example: '|L\d:' for the longform date format - note: the default format is Longform
+ * \:c - for a currency format
+ * \:<some-svg-key>|<some-optional-class-name>\: or \:<some-svg-key>\: - the latter skips the class name
+ * \:*<string-to-be-bolded>\:* - this warps a bolding span around a string - cam be anywhere withing a large string
+ */
+export const formatString = (string: string, SvgComponent: Component) => {
```

Comment:
i don't think the component should do this formatting- this is an implementation detail that the user of the component should implement (and is often project + api specific).

I don't think this type of super-duper-opinionated formatting belongs in a library like this one...

Reply:
fair observation - this requires that the formatter be passed into the `customFormatter` option in the 'headings' prop